source/code/source/playfab/PlayFabAndroidHttpPlugin.cpp

```diff
@@ -653,15 +674,15 @@ namespace PlayFab
         return false;
     }
 
-    void PlayFabAndroidHttpPlugin::ProcessResponse(RequestTask& requestTask, const int httpCode)
+    void PlayFabAndroidHttpPlugin::ProcessResponse(RequestTask& requestTask, const int httpCode, const std::string& requestId)
     {
         CallRequestContainer& requestContainer = requestTask.RequestContainer();
         Json::CharReaderBuilder jsonReaderFactory;
         std::unique_ptr<Json::CharReader> jsonReader(jsonReaderFactory.newCharReader());
         JSONCPP_STRING jsonParseErrors;
         const bool parsedSuccessfully = jsonReader->parse(requestContainer.responseString.c_str(), requestContainer.responseString.c_str() + requestContainer.responseString.length(), &requestContainer.responseJson, &jsonParseErrors);
 
-        requestContainer.errorWrapper.RequestId = requestContainer.GetRequestId();
+        requestContainer.errorWrapper.RequestId = requestId;
```

Comment:
requestContainer.SetRequestID(reqId)
makes this line change un-necessary

Reply:
Surprisingly, this is still a necessary change. The requestContainer is not the errorWrapper, without this line, the PlayFabError does not get the RequestId. 

---
In reply to: [357435817](https://github.com/PlayFab/XPlatCoreTemplate/pull/63#discussion_r357435817) [](ancestors = 357435817)

Reply:
Updated SetRequestId to also set the container. NOW this line is no longer required (which also looked so weird to have an object set something to itself).

---
In reply to: [357439799](https://github.com/PlayFab/XPlatCoreTemplate/pull/63#discussion_r357439799) [](ancestors = 357439799,357435817)